onnxruntime/test/optimizer/nhwc_optimizer_test.cc

```diff
@@ -0,0 +1,249 @@
+
+    for (auto& dim : shape) {
+      tensor_proto.add_dims(dim);
+    }
+
+    tensor_proto.mutable_float_data()->Resize(static_cast<int>(data.size()), 0.0f);
+    memcpy(tensor_proto.mutable_float_data()->mutable_data(), data.data(), data.size() * sizeof(float));
+
+    graph_.AddInitializedTensor(tensor_proto);
+
+    return &graph_.GetOrCreateNodeArg(name, nullptr);
+  }
+
+  NodeArg* MakeInitializer(const std::vector<int64_t>& shape) {
+    int64_t num_elements = std::accumulate(shape.begin(), shape.end(), int64_t(1), std::multiplies<int64_t>{});
+    return MakeInitializer(shape, FillRandomData(static_cast<size_t>(num_elements)));
+  }
+
+  Node& AddNode(const std::string& op_type,
+                const std::vector<NodeArg*>& input_args,
+                const std::vector<NodeArg*>& output_args) {
+   Node& node = graph_.AddNode(graph_.GenerateNodeName("node"),
+                          op_type,
+                          "description",
+                          input_args,
+                          output_args,
+                          nullptr,
+                          kOnnxDomain);
+   node.SetExecutionProviderType(kAclExecutionProvider);
+   return node;
+  }
+
+  Node& AddConvNode(NodeArg* input_arg, NodeArg* output_arg, const std::vector<int64_t>& weights_shape, bool no_bias = false) {
+    auto* weights_arg = MakeInitializer(weights_shape);
+    std::vector<NodeArg*> input_args = {input_arg, weights_arg};
+    if (!no_bias) {
+      auto* biases_arg = MakeInitializer({weights_shape[0]});
+      input_args.push_back(biases_arg);
+    }
+
+    return AddNode("Conv", input_args, {output_arg});
+  }
+
+  std::vector<float> FillRandomData(size_t count) {
+    constexpr int min_fill_value = -23;
+    constexpr int max_fill_value = 23;
+
+    std::vector<float> random_data {};
+    random_data.resize(count);
+
+    for (size_t n = 0; n < count; n++) {
+      random_data[n] = static_cast<float>(fill_value_);
+      fill_value_++;
+      if (fill_value_ == max_fill_value) {
+        fill_value_ = min_fill_value;
+      }
+    }
+
+    return random_data;
+  }
+
+  Graph& graph_;
+  NameMLValMap feeds_;
+  std::vector<std::string> output_names_;
+  int fill_value_;
+};
+
+void NhwcOptimizerTester(const std::function<void(NhwcTestHelper& helper)>& build_test_case,
+                          const std::function<void(NhwcInferenceSession& session)>& check_nhwc_graph,
+                          int opset_version = 10) {
+
+#ifndef USE_ACL
+  return;
+#endif
+
+  // Build the model for this test.
+  std::unordered_map<std::string, int> domain_to_version;
+  domain_to_version[kOnnxDomain] = opset_version;
+  Model model("nhwc", false, ModelMetaData(), PathString(), IOnnxRuntimeOpSchemaRegistryList(),
+              domain_to_version, {}, DefaultLoggingManager().DefaultLogger());
+
+  NhwcTestHelper helper(model.MainGraph());
+  build_test_case(helper);
+  ASSERT_TRUE(model.MainGraph().Resolve().IsOK());
+
```

Comment:
nit: prefer ASSERT_STATUS_OK as that will output any error message automatically

Reply:
Will change this